Client/game_sa/CPoolsSA.cpp

```diff
@@ -143,67 +140,25 @@ CVehicle* CPoolsSA::AddVehicle(DWORD* pGameInterface)
     return pVehicle;
 }
 
-void CPoolsSA::RemoveVehicle(unsigned long ulID, bool)
-{
-    DEBUG_TRACE("void CPoolsSA::RemoveVehicle ( unsigned long ulID, bool )");
-
-    static bool bIsDeletingVehicleAlready = false;            // to prevent delete being called twice
-    if (!bIsDeletingVehicleAlready)
-    {
-        bIsDeletingVehicleAlready = true;
-
-        CVehicleSA* pVehicleSA = m_vehiclePool.array[ulID];
-        assert(NULL != pVehicleSA);
-
-        // Pop the element to remove from the pool array
-        if (ulID != m_vehiclePool.ulCount - 1)
-        {
-            // We are removing an intermediate position of
-            // the array. Move the last element to the just
-            // deleted element position to not allow empty
-            // spaces on it.
-            m_vehiclePool.array[ulID] = m_vehiclePool.array[m_vehiclePool.ulCount - 1];
-            m_vehiclePool.array[ulID]->SetArrayID(ulID);
-        }
-        m_vehiclePool.array[m_vehiclePool.ulCount - 1] = NULL;
-
-        // Unlink the element to remove from the pool map
-        vehiclePool_t::mapType::iterator iter = m_vehiclePool.map.find(pVehicleSA->GetVehicleInterface());
-        if (iter != m_vehiclePool.map.end())
-        {
-            m_vehiclePool.map.erase(iter);
-        }
-
-        // Delete it from memory
-        delete pVehicleSA;
-
-        // Decrease the count of elements in the pool
-        --m_vehiclePool.ulCount;
-
-        bIsDeletingVehicleAlready = false;
-    }
-}
-
 void CPoolsSA::RemoveVehicle(CVehicle* pVehicle, bool bDelete)
 {
     DEBUG_TRACE("void CPoolsSA::RemoveVehicle ( CVehicle * pVehicle, bool bDelete )");
 
     assert(NULL != pVehicle);

```

Comment:
Add this check here:
```
static bool bIsDeletingVehicleAlready = false;            // to prevent delete being called twice
if (!bIsDeletingVehicleAlready)
{
    bIsDeletingVehicleAlready = true;
```